backend/src/handlers/user_handler.go

```diff
@@ -157,6 +157,17 @@ func (srv *Server) HandleCreateUser(w http.ResponseWriter, r *http.Request) {
 	if user.User.FacilityID == 0 {
 		user.User.FacilityID = srv.getFacilityID(r)
 	}
+	userNameExists := srv.Db.GetUserByUsername(user.User.Username)
```

Comment:
then in both the handlers we will just need

```go
if !database.IsValidUsername() {
  //...
}
```

This fixes the soft-delete issue, and prevents making several database queries and returning a whole row to make the check.

Reply:
Made those changes.  Thank you the tips, especially the mention about soft deletions - I learned some new things today :)

Reply:
Much better :+1: 

This is good with me if you are all set @calisio || @jtucholski

I will say that rebasing this is going to probably be a pain in the ass :frowning_face: depending on whether you have rebased from main once already to pull the formatting in and then made more changes in the process. Then it will be a complete nightmare and you may want to just scratch the branch and copy the changes over to a new branch you base off main, and then force push over this.